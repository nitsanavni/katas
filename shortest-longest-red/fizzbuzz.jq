def spec:
    [
        { d: 3, code: "Fizz" },
        { d: 5, code: "Buzz" },
        { d: 7, code: "Whizz" },
        { d: 11, code: "Bang" }
    ];

def default_to(n):
    if length == 0 then n else . end;

def gather_all_relevant_codes_for(n):
    spec | map(select(n % .d == 0) | .code) | join("");

def fizzbuzz:
    . as $n |
    gather_all_relevant_codes_for($n) |
    default_to($n);

def least_common_multiple:
    spec | map(.d) | reduce .[] as $d (1; . * $d);

range(least_common_multiple) | . + 1 | fizzbuzz
