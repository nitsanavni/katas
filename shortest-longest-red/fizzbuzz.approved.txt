1
2
"Fizz"
4
"Buzz"
"Fizz"
"Whizz"
8
"Fizz"
"Buzz"
"Bang"
"Fizz"
13
"Whizz"
"FizzBuzz"
16
17
"Fizz"
19
"Buzz"
"FizzWhizz"
"Bang"
23
"Fizz"
"Buzz"
26
"Fizz"
"Whizz"
29
"FizzBuzz"
31
32
"FizzBang"
34
"BuzzWhizz"
"Fizz"
37
38
"Fizz"
"Buzz"
41
"FizzWhizz"
43
"Bang"
"FizzBuzz"
46
47
"Fizz"
"Whizz"
"Buzz"
"Fizz"
52
53
"Fizz"
"BuzzBang"
"Whizz"
"Fizz"
58
59
"FizzBuzz"
61
62
"FizzWhizz"
64
"Buzz"
"FizzBang"
67
68
"Fizz"
"BuzzWhizz"
71
"Fizz"
73
74
"FizzBuzz"
76
"WhizzBang"
"Fizz"
79
"Buzz"
"Fizz"
82
83
"FizzWhizz"
"Buzz"
86
"Fizz"
"Bang"
89
"FizzBuzz"
"Whizz"
92
"Fizz"
94
"Buzz"
"Fizz"
97
"Whizz"
"FizzBang"
"Buzz"
101
"Fizz"
103
104
"FizzBuzzWhizz"
106
107
"Fizz"
109
"BuzzBang"
"Fizz"
"Whizz"
113
"Fizz"
"Buzz"
116
"Fizz"
118
"Whizz"
"FizzBuzz"
"Bang"
122
"Fizz"
124
"Buzz"
"FizzWhizz"
127
128
"Fizz"
"Buzz"
131
"FizzBang"
"Whizz"
134
"FizzBuzz"
136
137
"Fizz"
139
"BuzzWhizz"
"Fizz"
142
"Bang"
"Fizz"
"Buzz"
146
"FizzWhizz"
148
149
"FizzBuzz"
151
152
"Fizz"
"WhizzBang"
"Buzz"
"Fizz"
157
158
"Fizz"
"Buzz"
"Whizz"
"Fizz"
163
164
"FizzBuzzBang"
166
167
"FizzWhizz"
169
"Buzz"
"Fizz"
172
173
"Fizz"
"BuzzWhizz"
"Bang"
"Fizz"
178
179
"FizzBuzz"
181
"Whizz"
"Fizz"
184
"Buzz"
"Fizz"
"Bang"
188
"FizzWhizz"
"Buzz"
191
"Fizz"
193
194
"FizzBuzz"
"Whizz"
197
"FizzBang"
199
"Buzz"
"Fizz"
202
"Whizz"
"Fizz"
"Buzz"
206
"Fizz"
208
"Bang"
"FizzBuzzWhizz"
211
212
"Fizz"
214
"Buzz"
"Fizz"
"Whizz"
218
"Fizz"
"BuzzBang"
221
"Fizz"
223
"Whizz"
"FizzBuzz"
226
227
"Fizz"
229
"Buzz"
"FizzWhizzBang"
232
233
"Fizz"
"Buzz"
236
"Fizz"
"Whizz"
239
"FizzBuzz"
241
"Bang"
"Fizz"
244
"BuzzWhizz"
"Fizz"
247
248
"Fizz"
"Buzz"
251
"FizzWhizz"
"Bang"
254
"FizzBuzz"
256
257
"Fizz"
"Whizz"
"Buzz"
"Fizz"
262
263
"FizzBang"
"Buzz"
"Whizz"
"Fizz"
268
269
"FizzBuzz"
271
272
"FizzWhizz"
274
"BuzzBang"
"Fizz"
277
278
"Fizz"
"BuzzWhizz"
281
"Fizz"
283
284
"FizzBuzz"
"Bang"
"Whizz"
"Fizz"
289
"Buzz"
"Fizz"
292
293
"FizzWhizz"
"Buzz"
296
"FizzBang"
298
299
"FizzBuzz"
"Whizz"
302
"Fizz"
304
"Buzz"
"Fizz"
307
"WhizzBang"
"Fizz"
"Buzz"
311
"Fizz"
313
314
"FizzBuzzWhizz"
316
317
"Fizz"
"Bang"
"Buzz"
"Fizz"
"Whizz"
323
"Fizz"
"Buzz"
326
"Fizz"
328
"Whizz"
"FizzBuzzBang"
331
332
"Fizz"
334
"Buzz"
"FizzWhizz"
337
338
"Fizz"
"Buzz"
"Bang"
"Fizz"
"Whizz"
344
"FizzBuzz"
346
347
"Fizz"
349
"BuzzWhizz"
"Fizz"
"Bang"
353
"Fizz"
"Buzz"
356
"FizzWhizz"
358
359
"FizzBuzz"
361
362
"FizzBang"
"Whizz"
"Buzz"
"Fizz"
367
368
"Fizz"
"Buzz"
"Whizz"
"Fizz"
373
"Bang"
"FizzBuzz"
376
377
"FizzWhizz"
379
"Buzz"
"Fizz"
382
383
"Fizz"
"BuzzWhizzBang"
386
"Fizz"
388
389
"FizzBuzz"
391
"Whizz"
"Fizz"
394
"Buzz"
"FizzBang"
397
398
"FizzWhizz"
"Buzz"
401
"Fizz"
403
404
"FizzBuzz"
"Whizz"
"Bang"
"Fizz"
409
"Buzz"
"Fizz"
412
"Whizz"
"Fizz"
"Buzz"
416
"Fizz"
"Bang"
419
"FizzBuzzWhizz"
421
422
"Fizz"
424
"Buzz"
"Fizz"
"Whizz"
428
"FizzBang"
"Buzz"
431
"Fizz"
433
"Whizz"
"FizzBuzz"
436
437
"Fizz"
439
"BuzzBang"
"FizzWhizz"
442
443
"Fizz"
"Buzz"
446
"Fizz"
"Whizz"
449
"FizzBuzz"
"Bang"
452
"Fizz"
454
"BuzzWhizz"
"Fizz"
457
458
"Fizz"
"Buzz"
461
"FizzWhizzBang"
463
464
"FizzBuzz"
466
467
"Fizz"
"Whizz"
"Buzz"
"Fizz"
472
"Bang"
"Fizz"
"Buzz"
"Whizz"
"Fizz"
478
479
"FizzBuzz"
481
482
"FizzWhizz"
"Bang"
"Buzz"
"Fizz"
487
488
"Fizz"
"BuzzWhizz"
491
"Fizz"
493
494
"FizzBuzzBang"
496
"Whizz"
"Fizz"
499
"Buzz"
"Fizz"
502
503
"FizzWhizz"
"Buzz"
"Bang"
"Fizz"
508
509
"FizzBuzz"
"Whizz"
512
"Fizz"
514
"Buzz"
"Fizz"
"Bang"
"Whizz"
"Fizz"
"Buzz"
521
"Fizz"
523
524
"FizzBuzzWhizz"
526
527
"FizzBang"
529
"Buzz"
"Fizz"
"Whizz"
533
"Fizz"
"Buzz"
536
"Fizz"
538
"WhizzBang"
"FizzBuzz"
541
542
"Fizz"
544
"Buzz"
"FizzWhizz"
547
548
"Fizz"
"BuzzBang"
551
"Fizz"
"Whizz"
554
"FizzBuzz"
556
557
"Fizz"
559
"BuzzWhizz"
"FizzBang"
562
563
"Fizz"
"Buzz"
566
"FizzWhizz"
568
569
"FizzBuzz"
571
"Bang"
"Fizz"
"Whizz"
"Buzz"
"Fizz"
577
578
"Fizz"
"Buzz"
"Whizz"
"Fizz"
"Bang"
584
"FizzBuzz"
586
587
"FizzWhizz"
589
"Buzz"
"Fizz"
592
593
"FizzBang"
"BuzzWhizz"
596
"Fizz"
598
599
"FizzBuzz"
601
"Whizz"
"Fizz"
604
"BuzzBang"
"Fizz"
607
608
"FizzWhizz"
"Buzz"
611
"Fizz"
613
614
"FizzBuzz"
"WhizzBang"
617
"Fizz"
619
"Buzz"
"Fizz"
622
"Whizz"
"Fizz"
"Buzz"
626
"FizzBang"
628
629
"FizzBuzzWhizz"
631
632
"Fizz"
634
"Buzz"
"Fizz"
"Whizz"
"Bang"
"Fizz"
"Buzz"
641
"Fizz"
643
"Whizz"
"FizzBuzz"
646
647
"Fizz"
"Bang"
"Buzz"
"FizzWhizz"
652
653
"Fizz"
"Buzz"
656
"Fizz"
"Whizz"
659
"FizzBuzzBang"
661
662
"Fizz"
664
"BuzzWhizz"
"Fizz"
667
668
"Fizz"
"Buzz"
"Bang"
"FizzWhizz"
673
674
"FizzBuzz"
676
677
"Fizz"
"Whizz"
"Buzz"
"Fizz"
"Bang"
683
"Fizz"
"Buzz"
"Whizz"
"Fizz"
688
689
"FizzBuzz"
691
692
"FizzWhizzBang"
694
"Buzz"
"Fizz"
697
698
"Fizz"
"BuzzWhizz"
701
"Fizz"
703
"Bang"
"FizzBuzz"
706
"Whizz"
"Fizz"
709
"Buzz"
"Fizz"
712
713
"FizzWhizz"
"BuzzBang"
716
"Fizz"
718
719
"FizzBuzz"
"Whizz"
722
"Fizz"
724
"Buzz"
"FizzBang"
727
"Whizz"
"Fizz"
"Buzz"
731
"Fizz"
733
734
"FizzBuzzWhizz"
736
"Bang"
"Fizz"
739
"Buzz"
"Fizz"
"Whizz"
743
"Fizz"
"Buzz"
746
"Fizz"
"Bang"
"Whizz"
"FizzBuzz"
751
752
"Fizz"
754
"Buzz"
"FizzWhizz"
757
758
"FizzBang"
"Buzz"
761
"Fizz"
"Whizz"
764
"FizzBuzz"
766
767
"Fizz"
769
"BuzzWhizzBang"
"Fizz"
772
773
"Fizz"
"Buzz"
776
"FizzWhizz"
778
779
"FizzBuzz"
"Bang"
782
"Fizz"
"Whizz"
"Buzz"
"Fizz"
787
788
"Fizz"
"Buzz"
"Whizz"
"FizzBang"
793
794
"FizzBuzz"
796
797
"FizzWhizz"
799
"Buzz"
"Fizz"
802
"Bang"
"Fizz"
"BuzzWhizz"
806
"Fizz"
808
809
"FizzBuzz"
811
"Whizz"
"Fizz"
"Bang"
"Buzz"
"Fizz"
817
818
"FizzWhizz"
"Buzz"
821
"Fizz"
823
824
"FizzBuzzBang"
"Whizz"
827
"Fizz"
829
"Buzz"
"Fizz"
832
"Whizz"
"Fizz"
"Buzz"
"Bang"
"Fizz"
838
839
"FizzBuzzWhizz"
841
842
"Fizz"
844
"Buzz"
"Fizz"
"WhizzBang"
848
"Fizz"
"Buzz"
851
"Fizz"
853
"Whizz"
"FizzBuzz"
856
857
"FizzBang"
859
"Buzz"
"FizzWhizz"
862
863
"Fizz"
"Buzz"
866
"Fizz"
"Whizz"
"Bang"
"FizzBuzz"
871
872
"Fizz"
874
"BuzzWhizz"
"Fizz"
877
878
"Fizz"
"BuzzBang"
881
"FizzWhizz"
883
884
"FizzBuzz"
886
887
"Fizz"
"Whizz"
"Buzz"
"FizzBang"
892
893
"Fizz"
"Buzz"
"Whizz"
"Fizz"
898
899
"FizzBuzz"
901
"Bang"
"FizzWhizz"
904
"Buzz"
"Fizz"
907
908
"Fizz"
"BuzzWhizz"
911
"Fizz"
"Bang"
914
"FizzBuzz"
916
"Whizz"
"Fizz"
919
"Buzz"
"Fizz"
922
923
"FizzWhizzBang"
"Buzz"
926
"Fizz"
928
929
"FizzBuzz"
"Whizz"
932
"Fizz"
934
"BuzzBang"
"Fizz"
937
"Whizz"
"Fizz"
"Buzz"
941
"Fizz"
943
944
"FizzBuzzWhizz"
"Bang"
947
"Fizz"
949
"Buzz"
"Fizz"
"Whizz"
953
"Fizz"
"Buzz"
956
"FizzBang"
958
"Whizz"
"FizzBuzz"
961
962
"Fizz"
964
"Buzz"
"FizzWhizz"
967
"Bang"
"Fizz"
"Buzz"
971
"Fizz"
"Whizz"
974
"FizzBuzz"
976
977
"Fizz"
"Bang"
"BuzzWhizz"
"Fizz"
982
983
"Fizz"
"Buzz"
986
"FizzWhizz"
988
989
"FizzBuzzBang"
991
992
"Fizz"
"Whizz"
"Buzz"
"Fizz"
997
998
"Fizz"
"Buzz"
"WhizzBang"
"Fizz"
1003
1004
"FizzBuzz"
1006
1007
"FizzWhizz"
1009
"Buzz"
"Fizz"
"Bang"
1013
"Fizz"
"BuzzWhizz"
1016
"Fizz"
1018
1019
"FizzBuzz"
1021
"Whizz"
"FizzBang"
1024
"Buzz"
"Fizz"
1027
1028
"FizzWhizz"
"Buzz"
1031
"Fizz"
1033
"Bang"
"FizzBuzz"
"Whizz"
1037
"Fizz"
1039
"Buzz"
"Fizz"
1042
"Whizz"
"Fizz"
"BuzzBang"
1046
"Fizz"
1048
1049
"FizzBuzzWhizz"
1051
1052
"Fizz"
1054
"Buzz"
"FizzBang"
"Whizz"
1058
"Fizz"
"Buzz"
1061
"Fizz"
1063
"Whizz"
"FizzBuzz"
1066
"Bang"
"Fizz"
1069
"Buzz"
"FizzWhizz"
1072
1073
"Fizz"
"Buzz"
1076
"Fizz"
"WhizzBang"
1079
"FizzBuzz"
1081
1082
"Fizz"
1084
"BuzzWhizz"
"Fizz"
1087
1088
"FizzBang"
"Buzz"
1091
"FizzWhizz"
1093
1094
"FizzBuzz"
1096
1097
"Fizz"
"Whizz"
"BuzzBang"
"Fizz"
1102
1103
"Fizz"
"Buzz"
"Whizz"
"Fizz"
1108
1109
"FizzBuzz"
"Bang"
1112
"FizzWhizz"
1114
"Buzz"
"Fizz"
1117
1118
"Fizz"
"BuzzWhizz"
1121
"FizzBang"
1123
1124
"FizzBuzz"
1126
"Whizz"
"Fizz"
1129
"Buzz"
"Fizz"
1132
"Bang"
"FizzWhizz"
"Buzz"
1136
"Fizz"
1138
1139
"FizzBuzz"
"Whizz"
1142
"Fizz"
"Bang"
"Buzz"
"Fizz"
1147
"Whizz"
"Fizz"
"Buzz"
1151
"Fizz"
1153
1154
"FizzBuzzWhizzBang"
